elastic/kibana
# [Uptime] [Bug] Fix spaces in filters

# Note to reviewers
Be sure that https://github.com/elastic/kibana/pull/37564 is merged before reviewing this PR.

## Summary

Resolves #37921.

In https://github.com/elastic/kibana/issues/37921 it was reported that adding spaces to our filter criteria was broken. This was because the query generated by `EuiSearchBar` created a `match_phrase` clause that we weren't supporting.

In an attempt to curb this sort of thing from continuing to happen, this patch attempts to allow generic clauses to be applied, filtering only `match` clauses targeting the `monitor.status` field. If you need more explanation as to why this is a concern for us feel free to post questions in this PR.

## Testing
0. Configure a Heartbeat monitor with a name like `icmp test`. Be sure to also add at least one other filter. Run Heartbeat.
1. Load the Uptime UI, find your monitor in the list.
2. Create a filter query for the monitor's name, like `monitor.name:"icmp test"`.
  ![image](https://user-images.githubusercontent.com/18429259/59065905-d193f380-887b-11e9-8d2f-941ff8802754.png)
3. Observe that only your monitor is in the monitors list.


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.


### For maintainers
